etna/pipeline/assemble_pipelines.py

```diff
@@ -0,0 +1,41 @@
+from typing import List
+from typing import Sequence
+from typing import Union
+
+from etna.models.base import BaseModel
+from etna.pipeline.pipeline import Pipeline
+from etna.transforms import Transform
+
+
+def assemble_pipelines(
+    models: Union[BaseModel, Sequence[BaseModel]],
+    transforms: Sequence[Transform],
```

Comment:
It should be able to work with cases such as 

```
assemble_pipelines(Catboost(), [TrendTransform(in_column="target"), [LagTransform(lags=[1,2,3], in_column="target"), LagTransform(lags=[2,3,4], in_column="target")]], [1, 2])
```

And as a result it should create the following two pipelines:

```
Pipeline(model=Catboost(), transforms=[TrendTransform(in_column="target"), LagTransform(lags=[1,2,3], in_column="target")], horizon=1)
Pipeline(model=Catboost(), transforms=[TrendTransform(in_column="target"), LagTransform(lags=[2,3,4], in_column="target")], horizon=2)
```